ykrt/src/mt.rs

```diff
@@ -1020,31 +1018,16 @@ impl MTThread {
     }
 
     /// If a trace is currently running, return a reference to its `CompiledTrace`.
-    pub(crate) fn running_trace(
-        &self,
-    ) -> (
-        Option<Arc<dyn CompiledTrace>>,
-        Option<Arc<dyn CompiledTrace>>,
-    ) {
+    pub(crate) fn running_trace(&self) -> Option<Arc<dyn CompiledTrace>> {
         match &*self.tstate.borrow() {
-            MTThreadState::Executing { ctr: ctr_arc, root } => {
-                let root_clone = root.as_ref().map(Arc::clone);
-                (Some(Arc::clone(ctr_arc)), root_clone)
-            }
-            _ => (None, None),
+            MTThreadState::Executing { ctr } => Some(Arc::clone(ctr)),
+            _ => None,
         }
     }
 
     /// Update the currently running trace: `None` means that no trace is running.
-    pub(crate) fn set_running_trace(
-        &self,
-        ctr: Option<Arc<dyn CompiledTrace>>,
-        root: Option<Arc<dyn CompiledTrace>>,
-    ) {
-        *self.tstate.borrow_mut() = match ctr {
-            Some(ctr) => MTThreadState::Executing { ctr, root },
-            None => MTThreadState::Interpreting,
-        };
+    pub(crate) fn set_running_trace(&self, ctr: Arc<dyn CompiledTrace>) {
```

Comment:
We've removed the ability to set no running trace (by passing `None`) with this function, but the docstring implies that's still possible.

Either the docsting needs updating, or the parameter should accept `Option<...>`.

Is there still just one call site to this code? Should we inline it?

Reply:
Removed the second clause of the sentence in the force push.